adaptive/src/main/java/com/google/accompanist/adaptive/TwoPane.kt

```diff
@@ -0,0 +1,411 @@
+    val splitX = when (layoutDirection) {
+        LayoutDirection.Ltr ->
+            if (offsetFromStart) {
+                splitOffsetPixel
+            } else {
+                layoutCoordinates.size.width - splitOffsetPixel
+            }
+        LayoutDirection.Rtl ->
+            if (offsetFromStart) {
+                layoutCoordinates.size.width - splitOffsetPixel
+            } else {
+                splitOffsetPixel
+            }
+    }
+    val splitWidthPixel = with(density) { splitWidth.toPx() }
+
+    SplitResult(
+        isHorizontalSplit = true,
+        splitArea = Rect(
+            left = splitX - splitWidthPixel / 2f,
+            top = 0f,
+            right = splitX + splitWidthPixel / 2f,
+            bottom = layoutCoordinates.size.height.toFloat(),
+        )
+    )
+}
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots horizontally.
+ *
+ * The split will be placed at the given [splitFraction] from start, with the given [splitHeight].
+ *
+ * This strategy is _not_ fold aware.
+ */
+public fun FractionVerticalTwoPaneStrategy(
+    splitFraction: Float,
+    splitHeight: Dp = 0.dp,
+): TwoPaneStrategy = TwoPaneStrategy { density, _, layoutCoordinates ->
+    val splitY = layoutCoordinates.size.height * splitFraction
+    val splitHeightPixel = with(density) { splitHeight.toPx() }
+
+    SplitResult(
+        isHorizontalSplit = false,
+        splitArea = Rect(
+            left = 0f,
+            top = splitY - splitHeightPixel / 2f,
+            right = layoutCoordinates.size.width.toFloat(),
+            bottom = splitY + splitHeightPixel / 2f,
+        )
+    )
+}
+
+/**
+ * Returns a [TwoPaneStrategy] that will place the slots horizontally.
+ *
+ * The split will be placed at [splitOffset] either from the top or bottom based on
+ * [offsetFromTop], with the given [splitHeight].
+ *
+ * This strategy is _not_ fold aware.
+ */
+public fun FixedOffsetVerticalTwoPaneStrategy(
+    splitOffset: Dp,
+    offsetFromTop: Boolean,
+    splitHeight: Dp = 0.dp,
+): TwoPaneStrategy = TwoPaneStrategy { density, _, layoutCoordinates ->
+    val splitOffsetPixel = with(density) { splitOffset.toPx() }
+    val splitY =
+        if (offsetFromTop) {
+            splitOffsetPixel
+        } else {
+            layoutCoordinates.size.height - splitOffsetPixel
+        }
+    val splitHeightPixel = with(density) { splitHeight.toPx() }
+
+    SplitResult(
+        isHorizontalSplit = false,
+        splitArea = Rect(
+            left = 0f,
+            top = splitY - splitHeightPixel / 2f,
+            right = layoutCoordinates.size.width.toFloat(),
+            bottom = splitY + splitHeightPixel / 2f,
+        )
+    )
+}
+
+/**
+ * Returns a [DelegateTwoPaneStrategy] that will delegate [TwoPaneStrategy.calculateSplitResult] to
+ * the [firstStrategy] when [useFirstStrategy] returns `true`. Otherwise, it will delegate to
+ * the [secondStrategy].
+ */
+public fun DelegateTwoPaneStrategy(
```

Comment:
I assume that for the hinge usecase we will have to handle this manually, and for the rest of the usecases I didn't manage to find ones that would be worth of a separate API.

What's the usecase where this is needed?

Reply:
I included this as an example of how you can compose strategies that I use in the sample, but maybe this doesn't belong in the API: for more custom strategies, the implementation is fairly straightforward. I'll remove this from the API and show off the composability in the sample.

Reply:
Removed!